website/admin/views/admin_user_devices.py

```diff
@@ -0,0 +1,101 @@
+from flask import render_template, flash, redirect, url_for
+
+import data_interface
+from admin import admin_site
+from admin.views.admin_user_forms import AddNewDeviceForm, SetThermostatTargetForm
+
+
+
+motion_triggers = [{'id': '00', 'name': 'When Motion is Detected'},
+                   {'id': '01', 'name': 'When No Motion is Detected'}]
+thermostat_triggers = [{'id': '000', 'name': 'When the temperature is above 22'},
+                       {'id': '1111', 'name': 'When temperature is below 15'}]
+light_triggers = [{'id': '0000', 'name': 'Lights are on for 4 hours'}]
+door_triggers = [{'id': 'opens', 'name': "Opens"}, {'id': 'closes', 'name': "Closes"}]
+
+
+groupactions = ['Turn On', 'Turn Off', 'Set Temperature']
+
+actions = {"door_sensor": ['Turn on', 'Turn Off', 'No Action'],
+           "light_switch": ['Turn Switch on', 'Turn Switch Off', 'No Action'],
+           "thermostat": ['Turn on', 'Turn Off', 'No Action', 'Modify Temperature'],
+           "motion_sensor": ['Turn on', 'Turn Off', 'No Action']}
+
+
+@admin_site.route('/devices')
+def show_devices():
+    form = AddNewDeviceForm()
+    devices = data_interface.get_user_default_devices()
+    rooms = data_interface.get_user_default_rooms()
+    rooms = sorted(rooms, key=lambda k: k['name'])
+    any_linked = False
+    any_unlinked = False
+    if devices:
+        for device in devices:
+            if device['room_id'] != None:
+                any_linked = True
+            elif device['room_id'] == None:
+                any_unlinked = True
+    #change from default to focal user
+    #test requires here to check if devices returns devices correctly
+    return render_template("admin/admin_devices.html", devices=devices, groupactions=groupactions, rooms=rooms, new_device_form=form, table1=any_unlinked, table2=any_linked)
+
+
+@admin_site.route('/devices/new', methods=['POST', 'GET'])
+def add_new_device():
+    form = AddNewDeviceForm()
+    if form.validate_on_submit():
+        data_interface.add_new_device(device_type=form.device_type.data, vendor="OWN",
```

Comment:
This function doesn't seem to accept a user as argument, so we should probably change that as well.